### ISSUE TITLE
[PERF] Async Reprojection Causes Poor Performance

### ISSUE THREAD

**Issue by Goofybud16** [ValveSoftware/SteamVR-for-Linux#240]
**Describe the bug**
When Async Reprojection is enabled, it causes performance issues in titles that hit the target framerate. Beat Saber, for example, normally maintains a smooth 120 FPS. When I turn on Motion Smoothing, it seems to cause the framreate to "pulse" between 120FPS and barely playable (feels like less than 30 FPS). When I disable async reprojection in my steamvr.vrsettings file, everything runs fine (except no async reprojection). It even effects the blank grid before a game loads.

**To Reproduce**
1. Make Sure Async Reprojection is enabled
2. Play a game that performs well such as Beat Saber.

**Expected behavior**
Async Reprojection doesn't effect titles that already hit the desired framerate (120 FPS), and doesn't cause worse performance.

**System Information (please complete the following information):**
- Distribution: Debian Testing
- SteamVR version: 1.7.12
- Steam client version: 1566431379
- Opted into Steam client beta?: No
- Graphics driver version:  driverInfo = Mesa 19.1.4 (LLVM 8.0.1)
- Gist for SteamVR System Information: https://gist.github.com/Goofybud16/b032e537339fa964e1909705bbc743e8

**Performance Data (optional, but very helpful):**
- CPU: R9 3900X
- GPU: Radeon VII
- Gist for perf top: https://gist.github.com/Goofybud16/939b1e075b26753024bfba53c4c8e2b0
- Link to gpuvis trace: https://drive.google.com/open?id=1AhtOfszitzeymBWvNlx9iLXPHZDoK5Su

**Screenshots**
In Beat Saber:
![image](https://user-images.githubusercontent.com/669915/64085672-0849a080-cce9-11e9-88b2-e77dd38f5753.png)
In The Empty Grid:
![image](https://user-images.githubusercontent.com/669915/64085712-3af39900-cce9-11e9-9de1-3dce8372be67.png)

In VRChat, which normally only maintains 60 FPS, enabling async reprojection seems to cause it to run between 40 and 60 FPS.

While testing the issue, I had VRChat hang on exit (as usual), and that seemed to prompt a bunch of 

```
 kernel:[284877.299532] Uhhuh. NMI received for unknown reason 3c on CPU 18.
 kernel:[284892.197493] Uhhuh. NMI received for unknown reason 2c on CPU 23.
```

across all CPUs, repeatedly every few seconds (and sometimes multiple times a second).

**Note:** Commenters who are also experiencing this issue are encouraged to include the "System Information" and "Perf Data" sections in their replies.

**Comment by Zamundaaa** [comment]
This really sounds like #226, although I don't have that in the empty grid, and some games are more or less not affected at all. Quite possibly related.

**Comment by Goofybud16** [comment]
Back in 1.7.3, I didn't have this issue, and async reprojection seemed to work great (other than #230).

I haven't reproduced #230 on 1.7.12, but that's because the framerate tanks make it painful to play long enough to get that to happen.

Although looking at it, I do seem to have the impossible frame times like #226.

**Comment by Goofybud16** [comment]
1.7.13 seems to have fixed my problem with Beat Saber. The empty loading grid (because I have SteamVR Home disabled) still stutters and has impossible frame times, however in-game seems to be fine. There is an occasional stutter in Beat Saber, but it isn't bad (barely noticeable). It almost feels like I'm getting a 1-2 frame old frame displayed every now and then instead of just new frames. I suspect it's related to #230.

I also tested VRChat and found that the game now performs well (previously 40-60 FPS w/ Async on 1.7.12, now 60-120 w/ Async).

Seems to me like this problem has been resolved.
